lua/ttt2/libraries/migrations.lua

```diff
@@ -0,0 +1,73 @@
+---
+-- A TTT2 version migrations library
+-- @author ZenBre4ker
+-- @module migrations
+
+if SERVER then
+	AddCSLuaFile()
+end
+
+migrations = {}
+migrations.folderPath = "terrortown/migrations/"
+migrations.databaseName = "ttt2_migrations"
+migrations.savingKeys = {
+	executedAt = {
+		typ = "string",
+		default = ""
+	}
+}
+
+sql.CreateSqlTable(migrations.databaseName, migrations.savingKeys)
+migrations.orm = orm.Make(migrations.databaseName)
+
+---
+-- Runs all missing forward gamemode migrations.
+-- @realm shared
+function migrations.Apply()
+	local files = file.Find(migrations.folderPath .. "*.lua", "LUA", "nameasc") or {}
+
+	local fileExecutionCounter = 0
+
+	-- Add Hook to OnLuaError to catch if migrations fail
+	local isSuccess = true
+	local hookName = "OnLuaError"
+	local hookIdentifier = "TTT2MigrationErrorCatcher"
+	hook.Add(hookName, hookIdentifier, function(error, realm, stack, name, id)
+		isSuccess = false
+	end)
```

Comment:
This hook seems to be pretty new. It didnt work on my half a year old server. But after an update it did. Not sure if we should use it or warn to update the server?

Reply:
Sure about that? Doesn't seem to be new judging by the wiki:
![image](https://github.com/TTT-2/TTT2/assets/13639408/c39dbf1f-7a08-42f3-8c00-e17f7e49b7e0)

Reply:
Weird. It didnt work for me before the update.